internal/server/data/grant.go

```diff
@@ -104,8 +99,9 @@ func GetGrant(tx ReadTxn, opts GetGrantOptions) (*models.Grant, error) {
 	switch {
 	case opts.ByID != 0:
 		query.B("AND id = ?", opts.ByID)
-	case opts.BySubject != "":
-		query.B("AND subject = ?", opts.BySubject)
+	case opts.BySubject.ID != 0:
```

Comment:
Looks like you should be requiring subject kind here?

Reply:
Good call, I was kind of lazy with these checks. I guess all of these should error if there is an ID set without a kind. The query it produces is still safe in that it matches nothing, but an explicit error will make it easier to debug when we make a mistake.